multires-lyrics-search/tests/test_multireslyrics.py

```diff
@@ -0,0 +1,172 @@
+__copyright__ = 'Copyright (c) 2020 Jina AI Limited. All rights reserved.'
+__license__ = 'Apache-2.0'
+
+import csv
+import itertools
+import json
+import os
+import sys
+import subprocess
+
+from jina.flow import Flow
+import pytest
+from jina.proto import jina_pb2
+
+TOP_K = 3
+INDEX_FLOW_FILE_PATH = 'flows/index.yml'
+QUERY_FLOW_FILE_PATH = 'flows/query.yml'
+PORT = 45678
+
+
+# TODO restructure project so we don't duplicate input_fn
+def input_fn():
+    lyrics_file = os.environ.get('JINA_DATA_FILE')
+    with open(lyrics_file, newline='', encoding='utf-8') as f:
+        reader = csv.reader(f)
+        for row in itertools.islice(reader, int(os.environ.get('JINA_MAX_DOCS'))):
+            if row[-1] == 'ENGLISH':
+                d = jina_pb2.Document()
+                d.tags['ALink'] = row[0]
+                d.tags['SName'] = row[1]
+                d.tags['SLink'] = row[2]
+                d.text = row[3]
+                yield d
+
+
+def config(tmpdir):
+    parallel = 2 if sys.argv[1] == 'index' else 1
+
+    os.environ.setdefault('JINA_MAX_DOCS', '100')
+    os.environ.setdefault('JINA_PARALLEL', str(parallel))
+    os.environ.setdefault('JINA_SHARDS', str(1))
+    os.environ.setdefault('JINA_WORKSPACE', str(tmpdir))
+    os.environ.setdefault('JINA_DATA_FILE', 'tests/data-index.csv')
+    os.environ.setdefault('JINA_PORT', str(PORT))
+
+    os.makedirs(os.environ['JINA_WORKSPACE'], exist_ok=True)
+    return
+
+
+def index_documents():
+    f = Flow().load_config(INDEX_FLOW_FILE_PATH)
+
+    with f:
+        f.index(input_fn)
+
+
+def call_api(url, payload=None, headers=None):
+    if headers is None:
+        headers = {'Content-Type': 'application/json; charset=utf-8'}
+    import requests
+
+    return requests.post(url, data=json.dumps(payload), headers=headers).json()
+
+
+def get_results(query, top_k=TOP_K):
+    return call_api(
+        f'http://0.0.0.0:{PORT}/api/search', payload={'top_k': top_k, 'data': [query]}
+    )
+
+
+def get_flow():
+    f = Flow().load_config(QUERY_FLOW_FILE_PATH)
+    f.use_rest_gateway()
+    return f
+
+
+@pytest.fixture
+def queries_and_expected_replies():
+    return [
```

Comment:
Looks very good, I would just test then the direct resulting `match` of the `query`. Like this the test would look complete for me.

Reply:
You mean re-adding the assertion on the `'match'`-level, apart from the chunk-level, as it was before?

Reply:
yes, the `multi-level` assertion. It may look a little redundant but is a test so better to make sure everything makes sense

Reply:
Done